src/clusterfuzz/_internal/bot/tasks/utasks/variant_task.py

```diff
@@ -144,24 +169,50 @@ def execute_task(testcase_id, job_type):
     security_flag = False
     crash_stacktrace_output = 'No crash occurred.'
 
-  if original_job_type == job_type:
+  # Regular case of variant analysis.
+  variant.status = status
+  variant.revision = revision
+  variant.crash_type = crash_type
+  variant.crash_state = crash_state
+  variant.security_flag = security_flag
+  variant.is_similar = is_similar
+  variant.platform = environment.platform().lower()
+
+  return uworker_io.UworkerOutput(
+      testcase=testcase,
+      variant=variant,
+      job_type=job_type,
+      original_job_type=original_job_type,
+      revision=revision,
+      crash_stacktrace_output=crash_stacktrace_output)
+
+
+def handle_build_setup_error(output):
+  testcase = data_handler.get_testcase_by_id(
+      output.uworker_input['testcase_id'])
+  data_handler.update_testcase_comment(
+      testcase, data_types.TaskState.ERROR,
+      f'Build setup failed with job: {output.uworker_input["testcase_id"]}')
+
+
+def utask_postprocess(output):
+  """Handle the output from utask_main."""
+  if environment.is_engine_fuzzer_job(output.testcase.job_type):
+    # Remove put() method to avoid updates. DO NOT REMOVE THIS.
+    output.testcase.put = lambda: None
```

Comment:
Is this specific to variant task, or something more general?

Reply:
It's specific to the task.

Reply:
Could we add a bit more details why variant needs this in the comment?